unit_tests/OrthoTree/helpers/aid_list.cpp

```diff
@@ -231,74 +232,84 @@ TEST(AidListTest, NumParticlesInOctantTest) {
-                                                {.octant                    = 0b000,
-                                                 .expected_lower_bound      = 0,
-                                                 .expected_upper_bound_excl = 0,
-                                                 .expected_upper_bound_incl = 0,
-                                                 .expected_total_particles  = total_num_particles},
-
-                                                // top-left octant
-                                                {.octant                    = 0b001,
-                                                 .expected_lower_bound      = 0,
-                                                 .expected_upper_bound_excl = 100,
-                                                 .expected_upper_bound_incl = 99,
-                                                 .expected_total_particles  = n_particles_per_proc},
-
-                                                // top-middle octant
-                                                {.octant                    = 0b011,
-                                                 .expected_lower_bound      = 100,
-                                                 .expected_upper_bound_excl = 200,
-                                                 .expected_upper_bound_incl = 199,
-                                                 .expected_total_particles  = n_particles_per_proc},
-
-                                                // left-middle octant
-                                                {.octant                    = 0b101,
-                                                 .expected_lower_bound      = 200,
-                                                 .expected_upper_bound_excl = 300,
-                                                 .expected_upper_bound_incl = 299,
-                                                 .expected_total_particles  = n_particles_per_proc},
-
-                                                // middle-middle octant
-                                                {.octant                    = 0b111,
-                                                 .expected_lower_bound      = 300,
-                                                 .expected_upper_bound_excl = 400,
-                                                 .expected_upper_bound_incl = 399,
-                                                 .expected_total_particles = n_particles_per_proc}};
-
         for (auto test_data : tests_to_run) {
             run_test(test_data);
         }
     }
+
+    for (auto test_data : tests_to_run) {
+        std::vector<morton_code> octants = {test_data.octant};
+
+        // this has always size one, so we can just take the first element in the returning vector
+        const size_t calc_result = aid_list.getNumParticlesInOctantsParallel(octants).front();
+        EXPECT_EQ(test_data.expected_total_particles, calc_result)
```

Comment:
we now additionally check that each rank gets the correct number of particles in its octant if we call it in parallel